src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1

```diff
@@ -56,6 +58,14 @@ class AutoBugLog {
         if ($LogBugsForUnmappedResourceVarExistInPolicy) {
             $this.LogBugsForUnmappedResource = $LogBugsForUnmappedResourceVarExistInPolicy.Value;
         }
+
+        #Get if UpdateBug is enabled for this resource and get controls in policy.
+        if([Helpers]::CheckMember($this.ControlSettings.BugLogging, "UpdateBug.ResourceType") -and [Helpers]::CheckMember($this.ControlSettings.BugLogging.UpdateBug, "ControlIds") -and [Helpers]::CheckMember($this.ControlSettings.BugLogging.UpdateBug, "UpdateBugFields") ) {
```

Comment:
Since updatebug is an array - instead check whether this property exists in CS and if it exists check for its count to store whether update bug is enabled or not